schema-manager/src/test/java/io/camunda/search/schema/SchemaManagerStartupIT.java

```diff
@@ -0,0 +1,117 @@
+
+@Testcontainers
+class SchemaManagerStartupIT {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SchemaManagerStartupIT.class);
+
+  private static final String CAMUNDA_TEST_IMAGE_NAME =
+      Optional.ofNullable(System.getenv("CAMUNDA_TEST_DOCKER_IMAGE"))
+          .orElse("camunda/camunda:SNAPSHOT");
+
+  private final AtomicBoolean isShutdown = new AtomicBoolean(false);
+
+  @AutoClose
+  private final ZeebeContainer camunda =
+      new ZeebeContainer(DockerImageName.parse(CAMUNDA_TEST_IMAGE_NAME))
+          .withEnv("CAMUNDA_DATABASE_URL", "http://test-elasticsearch:9200")
+          .withEnv("SPRING_PROFILES_ACTIVE", "broker")
+          .withNetwork(Network.SHARED)
+          .waitingFor(
+              new WaitStrategy() {
+                @Override
+                public void waitUntilReady(final WaitStrategyTarget waitStrategyTarget) {
+                  // Wait until the application is shutdown
+                  Awaitility.await()
+                      .timeout(Duration.ofSeconds(60))
+                      .pollDelay(Duration.ofSeconds(1))
+                      .untilTrue(isShutdown);
+                }
+
+                @Override
+                public WaitStrategy withStartupTimeout(final Duration startupTimeout) {
+                  return this;
+                }
+              });
+
+  @Container
+  private final ElasticsearchContainer es =
+      createDefeaultElasticsearchContainer()
+          // Enable security in ES for to trigger the retry logic as application does not have
+          // credentials
+          .withEnv("xpack.security.enabled", "true")
+          .withNetwork(Network.SHARED)
+          .withNetworkAliases("test-elasticsearch");
+
+  @AutoClose private final ExecutorService executor = Executors.newSingleThreadExecutor();
+
+  @ParameterizedTest
+  @ValueSource(booleans = {false, true})
+  // false is to test the case when the shutdown is before the schema startup is started
+  // true is to test the case when the shutdown is when the schema startup is retrying
+  void shouldGracefullyShutdownWhenSchemaStartupStillRunning(final boolean waitForSchemaStartup)
+      throws InterruptedException {
+    // given
+    executor.submit(() -> camunda.start());
+
+    final var logToWaitFor =
+        waitForSchemaStartup
+            ? "Initializing search engine schema"
+            : "Root WebApplicationContext: initialization completed";
+
+    // just wait until the container is running (but not ready)
+    Awaitility.await()
+        .atMost(Duration.ofSeconds(60))
+        .until(() -> camunda.isRunning() && camunda.getLogs().contains(logToWaitFor));
+    // Let the startup proceed a little
+    Thread.sleep(2_000);
+
+    // when
+    // Simulate external shutdown signal
+    LOG.info("Start graceful shutdown of the container");
+    camunda.shutdownGracefully(Duration.ofSeconds(30)); // as default Kubernetes shutdown timeout
+    isShutdown.set(true);
```

Comment:
:wrench: If I understood correctly, you're using this flag as a latch. Why not just use a `CountdownLatch` directly? I know it will block the container startup thread, but that's probably fine? We're already doing that anyway. Is it because you want it to time out? You can also use `CountdownLatch#await(long, TimeUnit)` to do that.